juju/model.py

```diff
@@ -31,6 +31,22 @@ def relation(self, relation_name):
             # ideally integrating the error catching with Juju's mechanisms.
             raise TooManyRelatedApps(relation_name, num_related, 1)
 
+    def relation_by_id(self, relation_name, relation_id):
+        def _norm(relation_id):
+            # Relation IDs can be in the form '{relation_name}:{id}', so we need to
+            # normalize them to just '{id}' for comparison.
```

Comment:
This feels a bit awkward, though, in the sense that this idiosyncrasy should remain outside of the API. Is that coming from the environment variable?  If so, let's block it on the way in and prevent our internal APIs from accepting the bogus data.

That said, we must check that juju itself doesn't consider the relation ID to _indeed_ contain the name of the relation. That would point in a different direction, as in such a case we shouldn't be stripping it out.

Reply:
Juju AFAICT always reports relation IDs as `{relation_name}:{id}` even though they're used internally as just integers and it always accepts either form.  I think most devs are used to including the relation name, though, since that's what Juju reports.